packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,175 @@
+/**
+ * Get the last modified hermes profile
+ * @param packageName
+ */
+function getLatestFile(packageName: string): string {
+  try {
+    const file = execSync(`adb shell run-as ${packageName} ls cache/ -tp | grep -v /$ | egrep '\.cpuprofile' | head -1
+        `);
+    return file.toString().trim();
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+function execSyncWithLog(command: string) {
+  if (logger.isVerbose()) {
+    logger.debug(`${command}\n`);
+  }
+
+  return execSync(command);
+}
+
+/**
+ * Get the package name of the running React Native app
+ * @param config
+ */
+function getPackageName(config: Config) {
+  const androidProject = config.project.android;
+
+  if (!androidProject) {
+    throw new CLIError(`
+  Android project not found. Are you sure this is a React Native project?
+  If your Android files are located in a non-standard location (e.g. not inside \'android\' folder), consider setting
+  \`project.android.sourceDir\` option to point to a new location.
+`);
+  }
+  const {manifestPath} = androidProject;
+  const androidManifest = fs.readFileSync(manifestPath, 'utf8');
+
+  let packageNameMatchArray = androidManifest.match(/package="(.+?)"/);
+  if (!packageNameMatchArray || packageNameMatchArray.length === 0) {
+    throw new CLIError(
+      'Failed to build the app: No package name found. Found errors in /src/main/AndroidManifest.xml',
+    );
+  }
+
+  let packageName = packageNameMatchArray[1];
+
+  if (!validatePackageName(packageName)) {
+    logger.warn(
+      `Invalid application's package name "${chalk.bgRed(
+        packageName,
+      )}" in 'AndroidManifest.xml'. Read guidelines for setting the package name here: ${chalk.underline.dim(
+        'https://developer.android.com/studio/build/application-id',
+      )}`,
+    );
+  }
+  return packageName;
+}
+
+/**
+ * Validates that the package name is correct
+ * @param packageName
+ */
+function validatePackageName(packageName: string) {
+  return /^[a-z][a-z0-9_]*(\.[a-z][a-z0-9_]*)+$/.test(packageName);
+}
+
+/**
+ * Pull and convert a Hermes tracing profile to Chrome tracing profile
+ * @param ctx
+ * @param dstPath
+ * @param fileName
+ * @param sourceMapPath
+ * @param raw
+ * @param generateSourceMap
+ */
+export async function downloadProfile(
+  ctx: Config,
+  dstPath: string,
+  fileName?: string,
+  sourcemapPath?: string,
+  raw?: boolean,
+  shouldGenerateSourcemap?: boolean,
+) {
+  try {
+    const packageName = getPackageName(ctx);
+
+    // if file name is not specified, pull the latest file from device
+    const file = fileName || (await getLatestFile(packageName));
+    if (!file) {
+      logger.error(
+        'There is no file in the cache/ directory. Did you record a profile from the developer menu?\n',
+      );
+      process.exit(1);
+    }
+
+    logger.info(`File to be pulled: ${file}\n`);
+
+    //if destination path is not specified, pull to the current directory
```

Comment:
Warning - nitpicking:

Comment on line 98 starts: `// if` (with space), while this one - does not `//if`. Some of the comments later in the document are `//If`. 

Let's stick to a single style if that's okay - I would suggest `// If` (space & uppercase) to keep it consistent with other places.